live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/EventProcessorTest.java

```diff
@@ -298,6 +302,42 @@ void shouldNotProcessWhenPredictionNodeEvent()
         then(deleteNodeCommandHandler).shouldHaveNoInteractions();
     }
 
+    @Test
+    void shouldProcessWhenPredictionIsConfirmed()
+    {
+        String parentId = "parent_id";
+        String descriptionProperty = "cm:description";
+        String predictedDescription = "predicted description";
+
+        given(mockEvent.getData()).willReturn(mock());
+        given(mockEvent.getData().getResource()).willReturn(mock());
+        given(mockEvent.getData().getResource().getNodeType()).willReturn(PREDICTION_NODE_TYPE);
```

Comment:
This is a neat way to create nested mocks :+1:

Reply:
yep super smart🤯 I think Krystian or Maciek invented that